### PR TITLE
feat: add safe version of some of TypeScript's private type guards

### PR DESCRIPTION
**Pull request by RebeccaStevens**
## PR Checklist

- [ ] Addresses an existing open issue
- [ ] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/ts-api-utils/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Based off #49 

Recreates the logic of bunch of type guards defined in TypeScript's private API.
The logic slightly differs from TypeScript's in-order to make the function "safe".

For example: TypeScript's `isExpression` is designed to work with partially emitted node while this version requires fully emitted nodes.

### CODE REVIEW

File: src/nodes/typeGuards/single.ts [hunk 1]
```diff
@@ -1,122 +1,245 @@
...
+export type NumberKeyword = ts.KeywordToken<ts.SyntaxKind.NumberKeyword>;
+export type ObjectKeyword = ts.KeywordToken<ts.SyntaxKind.ObjectKeyword>;
+export type StringKeyword = ts.KeywordToken<ts.SyntaxKind.StringKeyword>;
+export type SuperKeyword = ts.KeywordToken<ts.SyntaxKind.SuperKeyword>;
+export type SymbolKeyword = ts.KeywordToken<ts.SyntaxKind.SymbolKeyword>;
+export type ThisKeyword = ts.KeywordToken<ts.SyntaxKind.ThisKeyword>;
+export type TrueKeyword = ts.KeywordToken<ts.SyntaxKind.TrueKeyword>;
+export type UndefinedKeyword = ts.KeywordToken<ts.SyntaxKind.UndefinedKeyword>;
+export type UnknownKeyword = ts.KeywordToken<ts.SyntaxKind.UnknownKeyword>;
+export type VoidKeyword = ts.KeywordToken<ts.SyntaxKind.VoidKeyword>;
```
Comment (JoshuaKGoldberg) [inline src/nodes/typeGuards/single.ts:19]: Duplicate changes from #49?

**Comment by JoshuaKGoldberg** [comment]
Marking as blocked on https://github.com/microsoft/TypeScript/issues/52727 - I'd like to see what the TS team is open to, in case we don't need to re-implement these

File: src/nodes/typeGuards/single.ts (hunk 1 above)
Comment (RebeccaStevens) [reply src/nodes/typeGuards/single.ts:19]: Yes. This pr is based of that one as it needs to use som of those functions.

**Comment by RebeccaStevens** [comment]
Closing for now as a lot of these are coming in TS 5.0

**Pull request closed without merging**